internal/build/phase_test.go

```diff
@@ -218,7 +242,9 @@ func testPhase(t *testing.T, when spec.G, it spec.S) {
 
 			when("#WithArgs", func() {
 				it("runs the subject phase with args", func() {
-					phase, err := subject.NewPhase("phase", build.WithArgs("some", "args"))
+					configProvider, err := build.NewPhaseConfigProvider(phaseName, lifecycle, build.WithArgs("some", "args"))
```

Comment:
Here we are transitively testing the implementation of `NewPhaseConfigProvider` through the implementation of our `phaseFactory` and the implementation of this phases `Run` method, relying on log output. We should look to add explicit tests to our `PhaseConfigProvider` that specifically test the implementation of our `With...` commands.

We can then hopefully drop a lot of the `#With...` contexts from this test suite.

Reply:
Adding tests for phase config provider. As mentioned in reply to other comment, we'll save a larger refactor of the test boundaries for `phase` for the future.